libs/domain/user/src/mixins/address.mixin.ts

```diff
@@ -1,44 +1,73 @@
 import { resolve, Type } from '@spryker-oryx/di';
+import { RouterService } from '@spryker-oryx/router';
 import {
   computed,
   signal,
   Signal,
   signalProperty,
 } from '@spryker-oryx/utilities';
 import { LitElement } from 'lit';
-import type { Address, AddressComponentProperties } from '../models';
-import { AddressService } from '../services';
+import { map } from 'rxjs';
+import type { Address, CrudState } from '../models';
+import { AddressService, AddressStateService } from '../services';
 
-export declare class AddressMixinInterface
-  implements AddressComponentProperties
-{
+export declare class AddressMixinInterface {
   protected addressService: AddressService;
-  addressId?: string;
-  address?: Address;
-  $address: Signal<Address | null>;
-  $addresses: Signal<Address[] | null>;
+  protected addressStateService: AddressStateService;
+
+  protected addressId: Signal<string | null>;
+  protected address: Signal<Address | null>;
+
+  /**
+   * Computed address Id, provided by a property, URL parameter or state.
+   */
+  protected $addressId: Signal<string | null>;
+
+  /**
+   * Computed address based on the current address. The current address is either
+   * dictated by the address id or the give data.
+   */
+  protected $address: Signal<Address | null>;
+
+  protected $action: Signal<CrudState>;
+
+  protected $addresses: Signal<Address[] | null>;
 }
 
-export const AddressMixin = <
-  T extends Type<LitElement & AddressComponentProperties>
->(
+export const AddressMixin = <T extends Type<LitElement>>(
   superClass: T
 ): Type<AddressMixinInterface> & T => {
   class AddressMixinClass extends superClass {
     protected addressService = resolve(AddressService);
+    protected addressStateService = resolve(AddressStateService);
+    protected routerService = resolve(RouterService);
 
     @signalProperty() addressId?: string;
     @signalProperty() address?: Address;
 
```

Comment:
`protected $isLoading  = signal(...add logic here...)`

Reply:
not yet sure.